### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by p1eDT**

### CODE REVIEW

**Review by CBuHuHa — APPROVED** [review]

File: Test/UiTests/LoginTests.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        //[Test]
+        //public void LoginWithEmptyUser()
+        //{
+        //    var user = UserBuilder.EmptyUser();
+        //    var loginPage = new LoginPage();
+        //    loginPage.TryToLogin(user);
+
+        //    Assert.That(loginPage.GetPopupMessage(), Is.EqualTo("Заполните это поле."));
+        //}
+    }
```
Comment (VladimirVolkovVD) [inline Test/UiTests/LoginTests.cs]: delete or uncomment

File: Test/UiTests/DeleteTestSuiteTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        //    var suites = new LoginPage()
+        //                .OpenPage()
+        //                .Login()
+        //                .SelectProject()
+        //                .OpenSuites()
+        //                .DeleteTestSuite(testSuiteForDelete);
+
+        //    Assert.Throws<NoSuchElementException>(() => suites.TestSuiteByName(testSuiteForDelete));
+        //}
```
Comment (VladimirVolkovVD) [inline Test/UiTests/DeleteTestSuiteTests.cs]: delete or uncomment

File: Test/UiTests/AddTestSuiteTests.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+        [Test]
+        [AllureTag("Positive tests")]
+        [AllureOwner("Nikita")]
+        [AllureSuite("TestMonitor")]
+        public void CreateTestSuiteTest()
+        {
+            Faker faker = new Faker();
+            string testSuiteName = faker.Commerce.ProductName();
+            string description = faker.Commerce.ProductDescription();
```
Comment (VladimirVolkovVD) [inline Test/UiTests/AddTestSuiteTests.cs]: create builder for it

File: Test/ApiTests/CreateTestCaseTests.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        public void UpdateTestCase()
+        {
+            int testSuiteId = 42;
+            logger.Message($"--Prepare: creation test case");
+
+            var oldtestCase = apiTestCaseSteps.CreateTestCase(testSuiteId);
+            logger.Message($"--Prepare: test case created with " +
+                $"\n code: {oldtestCase.Code} " +
+                $"\n name: {oldtestCase.Name} " +
+                $"\n id {oldtestCase.Id}");
```
Comment (VladimirVolkovVD) [inline Test/ApiTests/CreateTestCaseTests.cs]: move it out from test level

File: Core/TestLog.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                        return method.Name;
+                    }
+                }
+            }
+            else
+            {
+                result = "stack is null";
+            }
+            return result;
+        }
```
Comment (VladimirVolkovVD) [inline Core/TestLog.cs]:
```suggestion
        private string GetCallerTestDescription()
        {
            if (new StackTrace(false).GetFrames() != null)
            {
                foreach (var frame in stack)
                {
                    var method = frame.GetMethod();
                    var testAttributes = method.GetCustomAttributes<TestAttribute>();

                    if (testAttributes != null && testAttributes.Any())
                    {
                        return method.Name;
                    }

                    IEnumerable<TestCaseAttribute> tcAttributes = method.GetCustomAttributes<TestCaseAttribute>();
                    if (tcAttributes != null && tcAttributes.Any())
                    {
                        return method.Name;
                    }
                }
            }
         
           return "stack is null";
           
        }
```

File: Core/Selenium/Elements/ButtonBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        public static Button AddButton() 
+        { 
+            return new Button("primary");
+        }
+
+        public static Button SuccessButton()
+        {
+            return new Button("success");
+        }
+    }
```
Comment (VladimirVolkovVD) [inline Core/Selenium/Elements/ButtonBuilder.cs]: whe we need this class?

File: BussinesObject/UI/Steps/TestCaseBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
+﻿using Bogus;
+using Core;
+
+namespace BussinesObject.UI.Steps
+{
+    public class TestCaseBuilder
+    {
+        public TestCaseModel GetRandomTestCaseModel(string testSuite)
+        {
+            Faker faker = new Faker();
```
Comment (VladimirVolkovVD) [inline BussinesObject/UI/Steps/TestCaseBuilder.cs]: create base builder with faker field

File: BussinesObject/UI/Pages/NewTestCaseModal.cs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            addButton.ClickElementViaJs();
+        }
+
+        public void CreateTestCase(string name, string time, int stepCount)
+        {
+            testCaseName.SetText(name);
+            duration.GetElement().SendKeys(time);
+
+            var inputFirst = Browser.Instance.Driver.FindElement(By.XPath("//div[@class='ProseMirror is-input']"));
+            inputFirst.SendKeys(faker.Name.JobDescriptor());
```
Comment (VladimirVolkovVD) [inline BussinesObject/UI/Pages/NewTestCaseModal.cs]: locators shoud be in field

File: BussinesObject/UI/Pages/NewTestCaseModal.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+            for (int i = 1; i < stepCount; i++)
+            {
+                plusButton.ClickElementViaJs();
+
+                var input = Browser.Instance.Driver.FindElement(By.XPath("//p[@data-placeholder]/parent::div[contains(@class,'is-input')]"));
```
Comment (VladimirVolkovVD) [inline BussinesObject/UI/Pages/NewTestCaseModal.cs]: same

File: BussinesObject/Api/Steps/ApiUserSteps.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+            Faker faker = new Faker();
+
+            var userModel = new CreateUserModel()
+            {
+                FirstName = $"{faker.Person.FirstName}",
+                LastName = $"{faker.Person.LastName}",
+                Email = $"{faker.Person.Email}",
+                Password = "P@ssw0rd!",
+                PasswordConfirmation = "P@ssw0rd!",
+                Admin = false
```
Comment (VladimirVolkovVD) [inline BussinesObject/Api/Steps/ApiUserSteps.cs]: store password in config or use faker if it's random password

**Review by VladimirVolkovVD — CHANGES_REQUESTED** [review]
fix comments

File: Test/UiTests/LoginTests.cs (hunk 1 above)
Comment (p1eDT) [reply Test/UiTests/LoginTests.cs]: uncommented

File: Test/UiTests/DeleteTestSuiteTests.cs (hunk 2 above)
Comment (p1eDT) [reply Test/UiTests/DeleteTestSuiteTests.cs]: uncommented

File: BussinesObject/UI/Pages/NewTestCaseModal.cs (hunk 8 above)
Comment (p1eDT) [reply BussinesObject/UI/Pages/NewTestCaseModal.cs]: Fixed

File: BussinesObject/UI/Pages/NewTestCaseModal.cs (hunk 9 above)
Comment (p1eDT) [reply BussinesObject/UI/Pages/NewTestCaseModal.cs]: Fixed

File: BussinesObject/Api/Steps/ApiUserSteps.cs (hunk 10 above)
Comment (p1eDT) [reply BussinesObject/Api/Steps/ApiUserSteps.cs]: get password from faker

File: Test/UiTests/AddTestSuiteTests.cs (hunk 3 above)
Comment (p1eDT) [reply Test/UiTests/AddTestSuiteTests.cs]: created

File: Test/ApiTests/CreateTestCaseTests.cs (hunk 4 above)
Comment (p1eDT) [reply Test/ApiTests/CreateTestCaseTests.cs]: move to step

File: BussinesObject/UI/Steps/TestCaseBuilder.cs (hunk 7 above)
Comment (p1eDT) [reply BussinesObject/UI/Steps/TestCaseBuilder.cs]: created base builder

File: Core/Selenium/Elements/ButtonBuilder.cs (hunk 6 above)
Comment (p1eDT) [reply Core/Selenium/Elements/ButtonBuilder.cs]: the structure of the buttons has been redesigned, moved to their classes

**Comment by p1eDT** [comment]
comments fixed

**Review by VladimirVolkovVD — APPROVED** [review]

**Pull request merged**